controllers/observability_controller.go

```diff
@@ -226,20 +227,21 @@ func (r *ObservabilityReconciler) InitializeOperand(mgr ctrl.Manager) error {
 	}
 
 	storageClassExists := false
-	storageClass := &storage.StorageClass{}
-	selector := client.ObjectKey{
-		Name: storageClassGp2,
+	cluster := client.ObjectKey{
+		Name: "cluster",
 	}
+	infra := &infrastructure.Infrastructure{}
 
-	err = apiReader.Get(context.Background(), selector, storageClass)
-	if err == nil {
-		// no error means that the storage class was found
+	err = r.Get(context.Background(), cluster, infra)
+
+	// Check the infrastructure, If its Libvirt that means it is running on crc so it will run without storage.
+	if err == nil && string(infra.Status.PlatformStatus.Type) != "Libvirt" {
 		storageClassExists = true
```

Comment:
Just thinking if it would make sense to rename this variable to something like `operatorRunningLocally` or something like that, as even if the Libervert storage class exists, we are taking that as the indicator that we are running the operator on CRC. WDYT?

Reply:
Completely misread this yesterday, pushed up those changes now, had to make some changes below due to the renaming of the variable.